web/resource-proxy/alloy-light.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+# Given a URL from the proxy config and an ID, look up the unit type
+
+class XmlParser {
+    public $results = array();
+
+    function __construct($f = "proxy.config") {
+    $file = dirname(__FILE__) . "/../../conf/$f";
+    $xml = file_get_contents($file);
+        $parser = xml_parser_create();
+        xml_set_object($parser, $this);
+        xml_set_element_handler($parser, "tagStart", "tagEnd");
+        xml_parse($parser, $xml);
+        xml_parser_free($parser);
+    }
+
+    function tagStart($parser, $name, $attrs) {
+        $attrs = array_change_key_case($attrs, CASE_LOWER);
+        $tag = array(strtolower($name) => $attrs);
+        array_push($this->results, $tag);
+    }
+
+    function tagEnd($parser, $name) {
+        $this->results[count($this->results)-2]['childrens'][] = $this->results[count($this->results)-1];
+        array_pop($this->results);
+    }
+}
+
+header("Content-Type: application/json");
+
+$id = isset($_GET['id']) ? $_GET['id'] : null;
+$url = isset($_GET['url']) ? $_GET['url'] : null;
+$design = "designs_streetLights";
+$join_attribute = "root.attributes_streetLightingUnitsUnitType.attributes_streetLightingUnitTypesDescription";
+
+$xmlParser = new XmlParser();
+$data = $xmlParser->results[0]['childrens'][0]['childrens'];
+foreach ($data as $serverurl) {
+    $serverurl = $serverurl['serverurl'];
+    if (stripos($serverurl['url'], $url) === 0) {
+        process($serverurl, $id);
+    }
+}
+
+function process($serverurl, $id) {
+    $token = $serverurl['token'];
+    $url = str_replace('layer', 'aqs/join', $serverurl['hostredirect']);
+    $url .= "?token=$token";
+
+    $params = [
+        "aqs" => [
+               "type" => "Join",
+               "properties" => [
+                   "collectionCode" => ["Live"],
+                   "dodiCode" => $design,
+                   "joinAttributes" => [$join_attribute],
+               ],
+               "children"=> [[
+                   "type" => "Equals",
+                   "children" => [
+                       ["type" => "ItemProperty", "properties" => ["itemPropertyName" => "itemId"]],
+                       ["type" => "AlloyId", "properties" => ["value" => [$id]]]
+                   ]
+               ]]
+           ]
+       ];
+
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);
+    curl_setopt($ch, CURLOPT_POST, true);
+    curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($params));
+    $response = curl_exec($ch);
+    $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    if ($httpcode != 200) {
+        print $response;
```

Comment:
Don't think printing the entire output is a good idea, just in case there's something in there that shouldn't be. Probably best to just return an empty 500, or if there is something valuable for debug maybe put this behind a debug flag that's only enabled in staging.